internal/operators/handler/handler.go

```diff
@@ -0,0 +1,134 @@
+	restoperators "github.com/openshift/assisted-service/restapi/operations/operators"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Handler implements REST API interface and deals with HTTP objects and transport data model.
+type Handler struct {
+	// operatorsAPI is responsible for executing the actual logic related to the operators
+	operatorsAPI operators.API
+	db           *gorm.DB
+	log          logrus.FieldLogger
+}
+
+// NewHandler creates new handler
+func NewHandler(operatorsAPI operators.API, log logrus.FieldLogger, db *gorm.DB) *Handler {
+	return &Handler{operatorsAPI: operatorsAPI, log: log, db: db}
+}
+
+// ListOperatorProperties Lists properties for an operator name.
+func (h *Handler) ListOperatorProperties(ctx context.Context, params restoperators.ListOperatorPropertiesParams) middleware.Responder {
+	log := logutil.FromContext(ctx, h.log)
+	properties, err := h.operatorsAPI.GetOperatorProperties(params.OperatorName)
+	if err != nil {
+		log.Errorf("%s operator has not been found", params.OperatorName)
+		return restoperators.NewListOperatorPropertiesNotFound()
+	}
+
+	return restoperators.NewListOperatorPropertiesOK().
+		WithPayload(properties)
+}
+
+// ListSupportedOperators Retrieves the list of supported operators.
+func (h *Handler) ListSupportedOperators(_ context.Context, _ restoperators.ListSupportedOperatorsParams) middleware.Responder {
+	return restoperators.NewListSupportedOperatorsOK().
+		WithPayload(h.operatorsAPI.GetSupportedOperators())
+}
+
+// ListOfClusterOperators Lists operators to be monitored for a cluster.
+func (h *Handler) ListOfClusterOperators(ctx context.Context, params restoperators.ListOfClusterOperatorsParams) middleware.Responder {
+	operatorsList, err := h.GetMonitoredOperators(ctx, params.ClusterID, params.OperatorName, h.db)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return restoperators.NewListOfClusterOperatorsOK().WithPayload(operatorsList)
+}
+
+// ReportMonitoredOperatorStatus Controller API to report of monitored operators.
+func (h *Handler) ReportMonitoredOperatorStatus(ctx context.Context, params restoperators.ReportMonitoredOperatorStatusParams) middleware.Responder {
+	err := h.UpdateMonitoredOperatorStatus(ctx, params.ClusterID, params.ReportParams.Name, params.ReportParams.Status, params.ReportParams.StatusInfo)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return restoperators.NewReportMonitoredOperatorStatusOK()
+}
+
+// GetMonitoredOperators retrieves list of monitored operators for a cluster
+func (h *Handler) GetMonitoredOperators(ctx context.Context, clusterID strfmt.UUID, operatorName *string, db *gorm.DB) (models.MonitoredOperatorsList, error) {
+	log := logutil.FromContext(ctx, h.log)
+	var where *gorm.DB
+	if operatorName == nil || *operatorName == "" {
+		where = db.Where("cluster_id = ?", clusterID)
+	} else {
+		where = db.Where("name = ? and cluster_id = ?", *operatorName, clusterID)
+	}
```

Comment:
Maybe concatenate the where queries? It would look nicer

Reply:
Replaced with a map-based conditions.